### PR TITLE
Fix bug for concurrent parsing expr with strings

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #26662

### CODE REVIEW

**Review by longjiquan — APPROVED** [review]
/lgtm

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/parser/planparserv2/utils.go [hunk 1]
```diff
@@ -381,3 +382,30 @@ func IsAlwaysTruePlan(plan *planpb.PlanNode) bool {
...
+			escapeChCount++
+			continue
+		}
+		if (literal[i] == '"' && escapeChCount%2 == 0) || (literal[i] == '\'' && escapeChCount%2 != 0) {
+			needReplaceIndex = append(needReplaceIndex, i)
+		}
+		escapeChCount = 0
+	}
+	needReplaceIndexLength := len(needReplaceIndex)
+	for i := needReplaceIndexLength - 1; i >= 0; i-- {
```
Comment (czs007) [inline internal/parser/planparserv2/utils.go:401]: Use strings.Builder to accelearte

**Pull request merged**